File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        n = len(nums) - k + 1
        ans = [0] * n
        for i in range(n):
            ans[i] = max(nums[i:i+k])
        return ans
        

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> cnt {
            {0,1}
        }; // s[0] = 0单独统计
        int ans = 0, s = 0;
        for (int n: nums){
            s += n;
            ans += cnt[s-k];
            cnt[s]++;
        }
        return ans;
    }
};